### PR TITLE
Refactor EmitLoopCommon to make it more amenable to future extensions

### PR DESCRIPTION
**Pull request by Krovatkin**
This PR paves the way for support more iterator types in for-in loops.

### CODE REVIEW

**Comment by Krovatkin** [comment]
@driazati We need to support tensor targets in for-in loops (https://github.com/pytorch/pytorch/issues/19314), however, while I was adding tensor targets,  I realized we keep piling up special cases in `EmitLoopCommon`. I think we should refactor `emitLoopCommon` to make it a bit more flexible and then add tensor targets.

File: torch/csrc/jit/script/compiler.cpp [hunk 1]
```diff
@@ -1308,44 +1308,27 @@ struct to_ir {
...
-        cond_val = graph->insertConstant(true, nullptr, range);
-      }
+      max_trip_count_val = materializeConstant(
+          std::numeric_limits<int64_t>::max(),
+          *graph,
+          range,
+          integral_constants);
     }
+
+    auto true_constant  = graph->insertConstant(true, nullptr, range);
```
Comment (driazati) [inline torch/csrc/jit/script/compiler.cpp]: We don't always want to insert the constant into the graph if a condition is provided right? This should be inside an `if`

**Review by driazati — APPROVED** [review]

File: torch/csrc/jit/script/compiler.cpp (hunk 1 above)
Comment (Krovatkin) [reply torch/csrc/jit/script/compiler.cpp]: @driazati I'm counting on DCE to get rid of it when we use user's condition. I think having `true_constant` helps readability a little bit, but I can definitely remove `true_constant` and insert it only when needed. How strongly do you feel about this?

Comment (driazati) [reply torch/csrc/jit/script/compiler.cpp]: You can keep `true_constant` and just remove the ternary so it only gets inserted in the else branch, I don't think coupling this to a pass (even one pretty much guaranteed to run like DCE) is a great precedent to set

**Pull request closed without merging**